tasks/tests/unit/test_cache_test_rollups_redis.py

```diff
@@ -16,7 +16,7 @@ def test_cache_test_rollups(self, mock_storage, transactional_db):
         repo = RepositoryFactory()
 
         redis = get_redis_connection()
-        storage_service = get_storage_client()
+        storage_service = shared.storage.get_appropriate_storage_service()
```

Comment:
I wonder whether its even possible to always pass the `repo_id` to all the calls to `get_appropriate_storage_service`?

In this very example, you are passing the `repo_id` in the code, but not in the test, so these are out of sync. For tests, this does not matter at all. But for other parts of the codebase, I believe it might matter quite a bit if the reader and writer side are out of sync and get a different storage service because of it.